### DISCUSSION TITLE
Newly user created album should be opened on create in admin owned album

### DISCUSSION THREAD

**Discussion by haivala** [LycheeOrg/Lychee#2296] [Bugs]
### Lychee version

master

### Which PHP version are you using?

PHP 8.2

### Detailed description of the problem

When I create album as a user in admin owned album I'm redirected to front page.
I think the newly created album should opened instead.

### Steps to reproduce the issue

1. create album as admin
2. share album with user
3. make album public by url
4. login as user
5. try to create album to this admin owned album
6. User is redirected to front page 
7. As a user I cannot see this newly created album in the admin owned album.
8. When I login as admin the album has been created but the owner is admin.

Is this expected? Should it be changed?

I have album structure
Album (Event) - Album (photographer) - Pictures
I achieve this now so that I let the users make the Event album where they state their name and as a admin I move the user owned album to the admin owned album, but I think it would be more usable the way I just tried and described as It would give the user all permissions to the albums they own and permissions would not be transferred to admin.

### Diagnostics [REQUIRED]

```
System Information
------------------
Lychee Version (git):                    master (57e3503) -- Up to date (unknown).
DB Version:                              5.1.2

composer install:                        --no-dev
APP_ENV:                                 production
APP_DEBUG:                               false
APP_URL:                                 set
APP_DIR:                                 default
LOG_VIEWER_ENABLED:                      true
LIVEWIRE_ENABLED:                        true

System:                                  Linux
PHP Version:                             8.2.15
PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
Timezone:                                Europe/Helsinki
Max uploaded file size:                  500M
Max post size:                           300M
Livewire chunk size:                     12.00 MB
Max execution time:                      0
MySQL Version:                           10.11.7-MariaDB-1:10.11.7+maria~ubu2204

exec() Available:                        yes
Imagick Available:                       1
Imagick Enabled:                         1
Imagick Version:                         1809
GD Version:                              bundled (2.1.0 compatible)
Number of foreign key:                   12 found.
```

### Browser & System [REQUIRED]

FF, Docker

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
Hummm, okay. I some possible reasons, but that highlight also some problems.

1. create album as admin :heavy_check_mark: 
2. share album with user :question: _**Do you give write access to this user? (Just to double check)**_
3. make album public by url. :heavy_check_mark: 
4. login as user. :heavy_check_mark: 
5. try to create album to this admin owned album :heavy_check_mark: 
6. User is redirected to front page :heavy_check_mark: 

> As a user I cannot see this newly created album in the admin owned album.
> When I login as admin the album has been created but the owner is admin.

Yes, we assume that the tree is fully owned by the same user. It makes it easier to manage.
For this reason the newly created album is owned by admin. And because it is not shared, the user who is usually redirected after creation inside the album is automatically booted to front.

I did not think of this possibility.

**The fix I can see is that when such occasion occur, an _access_permission_ should also be added for the current user.**
Very easy fix actually.

**Reply by haivala** [reply]
Yes, I gave all sharing permissions.

**Comment by haivala** [comment]
Is there some special album which I could use to achieve the same thing?
`Album (Event) - Album (photographer) - Pictures`
The folder ownership is the problem in my setup now.

**Reply by ildyria** [reply]
Nah, with the access_permission change that should be easily fixed.
Give me a bit of time to fix it. :)

**Reply by ildyria** [reply]
@haivala Fixed on `alpha` branch.

IDK when it will be available on `master` (as it needs to be reviewed).

**Comment by haivala** [comment]
Problem with this approach is that the admin needs to setup the album and give permissions and in my situation there is not many persons who knows how to do that or can even login as admin. This means the photographers need to wait to be able to even create their album and after upload they loose the ownership of their album and pics.

Special album would be better fit for this. I'm just thinking out loud now: Root album that is automatically shared to every user and admin and everyone could make one (owner would be null). Everyone would be able to link their album inside this one and make it public that way. They could define the link name. The album tree would ignore sharing settings of the linked albums and just use the root albums sharing settings. To check the linked album sharing settings the urls could be something like `/gallery/(album name)/(linked album hash)/(pic hash)`. Basically this would be named public albums in easy to remember url format.

**Reply by haivala** [reply]
to make sure there would not be conflict on basic albums there could restriction on album name that it could not be same length as the album hash

**Reply by ildyria** [reply]
Once you set up the "root" album with "editing rights" for the selected users, they can create their sub-album and will have nearly all powers to that album.

**Yes they don't have 100% ownership, but the access permission allows them to edit, delete, move the pictures.**
The only access rights they don't have is transferring the ownership and changing the sharing parameters
(that granularity could be considered later).

**Reply by haivala** [reply]
if they would access their linked album on that shared album url the edit, delete and move options could be hidden. It could be used purely to make linked albums and their pics public

**Reply by ildyria** [reply]
> Special album would be better fit for this. I'm just thinking out loud now: Root album that is automatically shared to every user and admin and everyone could make one (owner would be null). Everyone would be able to link their album inside this one and make it public that way. They could define the link name. The album tree would ignore sharing settings of the linked albums and just use the root albums sharing settings. To check the linked album sharing settings the urls could be something like /gallery/(album name)/(linked album hash)/(pic hash). Basically this would be named public albums in easy to remember url format.

Maybe we could consider a new concept: _Category_ instead of root albums. A category would allow to create _sub_ albums with pre-defined permissions. :|

**Reply by haivala** [reply]
Yeah, category could be one option

**Reply by haivala** [reply]
Category or sharing album

**Reply by haivala** [reply]
Sharing album would be more descriptive
